pkg/controller/controller_utils.go

```diff
@@ -54,6 +54,13 @@ var (
 	KeyFunc = framework.DeletionHandlingMetaNamespaceKeyFunc
 )
 
+type ResyncPeriodFunc func() time.Duration
```

Comment:
It makes sense that you want a different value for every controller, but since each controller only calls the function once (when it makes the informer) it still doesn't make sense to me that you need to inject a function.